site/public/js/gradeable-notebook.js

```diff
@@ -94,43 +130,54 @@ function saveNotebookToLocal() {
  */
 function restoreNotebookFromLocal() {
     if (typeof autosaveEnabled !== 'undefined' && autosaveEnabled) {
-        const state = JSON.parse(localStorage.getItem(notebookAutosaveKey()));
+        const inputs = JSON.parse(localStorage.getItem(notebookAutosaveKey()));
 
-        if (state === null) {
+        if (inputs === null) {
             return;
         }
-
-        // First, we restore multiple choice answers
-        for (const id in state.multiple_choice) {
-            const values = state.multiple_choice[id];
-            // Extract the index from the ID generated from gatherInputAnswersByType()
-            // Index is stored after multiple_choice_, so substring it out
-            const index = id.substring('multiple_choice_'.length);
-            $(`#mc_field_${index} :input`).each((_index, element) => {
-                $(element).prop('checked', values.includes(element.value)).change();
+        //to prevent data loss when these changes get installed on production
+        const not_found = [];
+
+        //restore multiple choice
+        for (const id in inputs.multiple_choice) {
+            const filename = inputs.multiple_choice[id][0];
+            const answers = inputs.multiple_choice[id][1];
+            let found = false;
+            $(`fieldset.mc[data-filename="${filename}"] input`).each(function() {
+                found = true;
+                //check off proper inputs
+                for (let i = 0; i < answers.length; i++) {
+                    if ($(this)[0].defaultValue === answers[i]) {
+                        $(this).prop('checked', true);
+                    }
+                }
             });
+            if (!found) {
+                not_found.push(answers);
+            }
         }
-        // Next, we restore short-answer boxes
-        for (const id in state.short_answer) {
-            const answer = state.short_answer[id][0];
-            // Restore the answer and trigger change events (for setting button states)
-            // (see https://stackoverflow.com/questions/4672505/why-does-the-jquery-change-event-not-trigger-when-i-set-the-value-of-a-select-us)
-            $(`#${id}`).val(answer).trigger('input');
+
+        //restore short answers
+        for (const id in inputs.short_answer) {
+            const filename = inputs.short_answer[id][0];
+            const answers = inputs.short_answer[id][1];
+            let found = false;
+            $(`.short_answer > div[data-filename="${filename}"]`).each(function(){
+                //set input
+                found = true;
+                const editor = ($(this)[0]).querySelector('.CodeMirror').CodeMirror;
```

Comment:
Same as before, can use find to skip this awkward conversion
```suggestion
                const editor = $(this).find('.CodeMirror').CodeMirror;
```

Reply:
For some reason I could only get this to work using the way I had it before, that's how it was done previously as well.